api/app.py

```diff
@@ -0,0 +1,68 @@
+from flask import Flask
+from flask import request
+from flask import url_for
+from geodepy.geodesy import vincdir
+from geodepy.geodesy import vincinv
+from geodepy.convert import dms2dd
+from geodepy.convert import dd2dms
+
+app = Flask(__name__)
+
+coord2dd = {
+    'dd': lambda x: x,
+    'dms': dms2dd,
+}
+
+dd2coord = {
+    'dd': lambda x: x,
+    'dms': dd2dms,
+}
+
+
+@app.route('/')
+def list_routes():
+    return str(tuple(url_for(rule.endpoint) for rule in app.url_map.iter_rules() if rule.endpoint != 'static'))
+
+
+@app.route('/vincinv')
+def handle_vincinv():
+    from_coord = request.args.get('from_coord')
+    to_coord = request.args.get('to_coord')
+    lat1 = request.args.get('lat1', type=float)
+    lon1 = request.args.get('lon1', type=float)
+    lat2 = request.args.get('lat2', type=float)
+    lon2 = request.args.get('lon2', type=float)
+
+    dd = coord2dd[from_coord]
+    lat1_dd, lon1_dd, lat2_dd, lon2_dd = dd(lat1), dd(lon1), dd(lat2), dd(lon2)
+
+    ell_dist_dd, azimuth1to2_dd, azimuth2to1_dd = vincinv(lat1_dd, lon1_dd, lat2_dd, lon2_dd)
+
+    coord = dd2coord[to_coord]
+    ell_dist, azimuth1to2, azimuth2to1 = coord(ell_dist_dd), coord(azimuth1to2_dd), coord(azimuth2to1_dd)
+
+    return str((ell_dist, azimuth1to2, azimuth2to1)), 200
```

Comment:
I'd personally prefer this to be a JSON return type. What do others think? I don't really like getting back what looks like a python triple as a string.

Reply:
I agree. Pure laziness on my part.

Reply:
Question is if we're going to respond in JSON, do we also support returning as CSV too?

Reply:
Why would you support csv response type in a world where you can only invoke the endpoint with one set of input params?

Reply:
The online web page needs to be able to return a csv (and submit data in csv format) as this is the easiest for lay users to read and write. Most users of the online web page won't be able to understand or interpret a JSON object. Users of the API calls however would prefer the results in JSON. Is it possible for the Web UI to handle JSON -> csv?

Reply:
Yeah, I think this is a pretty typical setup. It also gives the front end the flexibility to run all of the queries asynchronously. Meaning your user doesn't have to wait for the entire file to be processed (for this reason it also reduces complexity in the API - long running jobs are a pain). The webpage would store these results in memory on the clients side, and they could then export to CSV for example.